### PR TITLE
Auto-generate docs for file descriptor and stdin sources

### LINKED ISSUE #15840: Convert the `file_descriptor` source to use auto-generated docs

### LINKED ISSUE #15846: Convert the `stdin` source to use auto-generated docs

### PR DESCRIPTION
**Pull request by davidhuie-dd**
Closes #15840, closes #15846.

The biggest change is that I inline ignored decoding and framing docs, because it looks like those are added magically somewhere. Also, `make serve` errors out due to duplicate entries for decoding and framing doc without that change.

### CODE REVIEW

File: src/sources/file_descriptors/file_descriptor.rs [hunk 1]
```diff
@@ -29,13 +30,16 @@ pub struct FileDescriptorSourceConfig {
     pub host_key: Option<OptionalValuePath>,
 
     #[configurable(derived)]
+    #[configurable(metadata(docs::hidden))]
```
Comment (neuronull) [inline src/sources/file_descriptors/file_descriptor.rs:33]: I'm actually not sure if this is necessary/correct/what to do  🤔 , @tobz  probably knows.

The `FramingConfig` is documented already in `lib/codecs/src/encoding/mod.rs` , so I'm wondering if adding this docs hidden change is actually a counter-productive thing.

Comment (neuronull) [reply src/sources/file_descriptors/file_descriptor.rs:33]: Oh I just re-read

> errors out due to duplicate entries for decoding and framing doc without that change.

hmmm 🤔 it _feels_ like it shouldn't be necessary to explicitly hide it if it's already documented where it's defined.

Comment (spencergilbert) [reply src/sources/file_descriptors/file_descriptor.rs:33]: You can add a `auto_generated: true` to https://github.com/vectordotdev/vector/pull/15855/files#diff-704bafbf43a586c4f12a330f7bd7503cb4bf05046437d6c4096f9e7a59d95752R15-R16, think that should resolve the problem

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: src/sources/file_descriptors/file_descriptor.rs (hunk 1 above)
Comment (tobz) [reply src/sources/file_descriptors/file_descriptor.rs:33]: Yeah, adding `auto_generated: true` is the correct solution.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**